SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialSpanIdStack.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialSpanIdStack.h"
+
+#include "Interop/Connection/SpatialEventTracer.h"
+
+namespace SpatialGDK
+{
+FSpatialSpanIdStack::FSpatialSpanIdStack(const SpatialEventTracer* InEventTracer)
+	: EventTracer(InEventTracer)
+{
+}
+
+void FSpatialSpanIdStack::SetEventTracer(const SpatialEventTracer* InEventTracer)
+{
+	EventTracer = InEventTracer;
+}
+
+void FSpatialSpanIdStack::AddNewLayer(const Trace_SpanId& SpanId)
+{
+	Stack.Add(SpanId);
+}
+
+void FSpatialSpanIdStack::AddToLayer(const Trace_SpanId& SpanId)
+{
+	const int32 Size = Stack.Num();
+	if (Size == 0)
+	{
+		AddNewLayer(SpanId);
+		return;
+	}
+
+	Trace_SpanId TopSpanId = Stack[Size - 1];
+	Trace_SpanId MergeCauses[2] = { SpanId, TopSpanId };
+	Stack[Size - 1] = EventTracer->CreateSpan(MergeCauses, 2).GetValue();
```

Comment:
What happens here if we say do 

Trace_SpanId a = CreateSpan(...);
Trace_SpanId b = CreateSpan(...);
Trace_SpanId c = CreateSpan([a, b]);
Trace_SpanId d = CreateSpan(...);
Trace_SpanId e = CreateSpan([c, d]);

We haven't logged any events for a, b or c, will this still generate good data from the trace tool? Might already have this answered in one of your tests

Reply:
I think there will just be SpanIds without events. All SpanIds will be causally linked though so you should be able to trace the data back. This is similar to what we do for component update except there we trace and even each time we merge. Maybe I will add an event just for consistency. We might want to just remove the event from component update though.